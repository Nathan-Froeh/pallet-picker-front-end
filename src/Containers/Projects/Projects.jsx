import React, { Component } from 'react';
import { postProject } from '../../ApiCalls';
import { Project } from '../Project/Project';
import './Projects.scss'

export class Projects extends Component {
  constructor() {
    super()
    this.state = {
      projectName: ''
    }
  }

  handleProjectName = (e) => {
    this.setState({projectName: e.target.value})
  }

  handleSubmitProject = (e) => {
    e.preventDefault()
    const name = {name: this.state.projectName}
    postProject(name)
  }

  render() {
    const addProjects = this.props.projects.map((project, key) => {
      const palettes = this.props.palettes.filter(palette => {
        return project.id === palette.project_id

      })
      console.log(palettes)
      return <Project name={project.name} id={project.id} palettes={palettes} key={key}/>
    })
    return (
      <div className='Projects'>
        <form onSubmit={this.handleSubmitProject}>
          <input 
            type="text" 
            placeholder="New project name" 
            onChange={this.handleProjectName}
          />
          <input type="submit"/>
        </form>
        <section className="project-container">
          {addProjects}
        </section>
      </div>
    )
  }
}

export default Projects
