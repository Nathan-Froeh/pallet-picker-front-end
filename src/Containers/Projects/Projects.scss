
.Projects {
  form {
    margin: 20px;
    input {
      height: 30px;
      border-radius: 5px;
      border: none;
      margin: 5px;
    }
  }
}
.project-container{
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}