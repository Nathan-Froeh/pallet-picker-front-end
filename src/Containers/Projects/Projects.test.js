import React from 'react';
import { shallow } from 'enzyme';
import Projects from './Projects';

describe('Projects', () => {
  let wrapper, instance;
  const props = {
    projects: [],
    name:'project name',
    id: 1,
    palettes: [],
    deleteProject: jest.fn(),
    deletePalette: jest.fn()
  }

  beforeEach(() => {
    wrapper = shallow(<Projects {...props}/>)
    instance = wrapper.instance()

  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change state when handleProjectName is invoked', () => {
    const mockEvent = { target: { newProjectName: 'title', value: 'Project Name'} };
    const expected = 'Project Name';
    wrapper.instance().handleProjectName(mockEvent);
    expect(wrapper.state('newProjectName')).toEqual(expected)
  })

  it('should run postProject when handleSubmitProject is called', () => {
    const currentState = {newProjectName:'Project Name'};
    wrapper.instance().setState(currentState);
    wrapper.instance().handleSubmitProject({preventDefault: jest.fn()});
    expect(wrapper.state('newProjectName')).toEqual(currentState.newProjectName)
  })
})