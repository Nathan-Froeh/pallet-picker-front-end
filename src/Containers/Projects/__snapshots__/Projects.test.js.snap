// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should match snapshot 1`] = `
<div
  className="Projects"
>
  <form
    id="new_project"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="New project name"
      required={true}
      type="text"
    />
    <input
      type="submit"
      value="Save project"
    />
  </form>
  <section
    className="project-container"
  />
</div>
`;
